flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SlotSharingExecutionSlotAllocatorTest.java

```diff
@@ -221,24 +204,26 @@ public void testReturningLogicalSlotsRemovesSharedSlot() throws Exception {
     }
 
     @Test
-    public void testLogicalSlotCancelsPhysicalSlotRequestAndRemovesSharedSlot() throws Exception {
+    void testLogicalSlotCancelsPhysicalSlotRequestAndRemovesSharedSlot() throws Exception {
         // physical slot request is not completed and does not complete logical requests
         testLogicalSlotRequestCancellationOrRelease(
                 true,
                 true,
                 (context, assignment) -> {
                     context.getAllocator().cancel(assignment.getExecutionAttemptId());
-                    try {
-                        assignment.getLogicalSlotFuture().get();
-                        fail("The logical future must finish with the cancellation exception");
-                    } catch (InterruptedException | ExecutionException e) {
-                        assertThat(e.getCause(), instanceOf(CancellationException.class));
-                    }
+                    assertThatThrownBy(
+                                    () -> {
+                                        context.getAllocator()
+                                                .cancel(assignment.getExecutionAttemptId());
+                                        assignment.getLogicalSlotFuture().get();
+                                    })
+                            .as("The logical future must finish with the cancellation exception.")
+                            .hasCauseInstanceOf(CancellationException.class);
```

Comment:
Maybe `hasRootCauseInstanceOf` ?

Reply:
They both can work. However, I think `hasCauseInstanceOf` is more accurate as a translation of the original logic.